Update change log and version for 5.9.0 beta 2 (#1219)
microsoft/msphpsql: CHANGELOG.md

The format is based on [Keep a Changelog](http://keepachangelog.com/)

## 5.9.0-beta2 - 2020-12-02
Updated PECL release packages. Here is the list of updates:

### Added
- Support for PHP 8.0

### Removed
- Dropped support for PHP 7.2

### Fixed

> - Pull Request [#1205](https://github.com/microsoft/msphpsql/pull/1205) - minimized compilation warnings in Linux and Mac 

Reviewer: ```suggestion
- Pull Request [#1205](https://github.com/microsoft/msphpsql/pull/1205) - minimized compilation warnings on Linux and macOS 
```

- Pull Request [#1209](https://github.com/microsoft/msphpsql/pull/1209) - fixed a bug in fetching varbinary max fields as char or wide chars

> - Issue [#1210](https://github.com/microsoft/msphpsql/issues/1210) - renamed the preview as beta to work with Pickle

Reviewer: ```suggestion
- Issue [#1210](https://github.com/microsoft/msphpsql/issues/1210) - Switched from preview to beta terminology to enable Pickle support
```
